### PR TITLE
Staticlib

### PR DESCRIPTION
**Pull request by mnunberg**

### CODE REVIEW

**Comment by mnunberg** [comment]
This PR also has some minor unrelated fixes; I can make them into separate PRs if you'd like.

**Comment by MeirShpilraien** [comment]
@mnunberg please notice that if we continue to use the RedisModuleApi even as a static library then we must call `RedisModule_Init` so the RedisModuleApi function will be initialized, but we must not register any commands or open any threads. I guess to right way to go here is to split the `RediSearch_Init` function to two functions, one that only calls `RedisModule_Init` and we should ask from the user to call this function on his init function. Also we should remove the check for busy module name and the call to `RedisModule_SetModuleAttribs` cause we do not want to expose that redisearch is actually loaded in redis.

File: src/module-init/module-init.c [hunk 1]
```diff
@@ -3,14 +3,159 @@ REDISMODULE_INIT_SYMBOLS();
...
+
+  sds confstr = RSConfig_GetInfoString(&RSGlobalConfig);
+  DO_LOG("notice", confstr);
+  sdsfree(confstr);
+
+  // Init extension mechanism
+  Extensions_Init();
+
+  if (RSGlobalConfig.concurrentMode) {
+    ConcurrentSearch_ThreadPoolStart();
```
Comment (MeirShpilraien) [inline src/module-init/module-init.c]: Do we need it on library?

File: src/module-init/module-init.c [hunk 2]
```diff
@@ -3,14 +3,159 @@ REDISMODULE_INIT_SYMBOLS();
...
+  }
+
+  // Init cursors mechanism
+  CursorList_Init(&RSCursors);
```
Comment (MeirShpilraien) [inline src/module-init/module-init.c]: Do we need it on library?

File: src/module-init/module-init.c (hunk 1 above)
Comment (mnunberg) [reply src/module-init/module-init.c]: I didn't want to complicate the code. The globals are de-initialized at module shutdown anyway, so I don't think it matters much..

Comment (MeirShpilraien) [reply src/module-init/module-init.c]: But it will open threads doesn't it?

Comment (mnunberg) [reply src/module-init/module-init.c]: Probably the better way to solve this is to simply set concurrentMode to off

Comment (MeirShpilraien) [reply src/module-init/module-init.c]: So lets set it if loaded as static lib

**Pull request merged**